From 6696ede19516cfaddec45fa9f543e9a29520b385 Mon Sep 17 00:00:00 2001
From: Eugene Susla <eugenesusla@google.com>
Date: Mon, 26 Oct 2020 09:08:49 -0700
Subject: [PATCH] Prevent non-system overlays from showing over CDM UI

Since CDM grants privileges, it should have the same overlay
policy as permission UI

Test: use an app wit ha visible overlay to ensure
the overlay disappears when CDM is shown
Fixes: 171221090
Change-Id: I004843edf5a156dc07fb961edf80272f5cb50163
(cherry picked from commit ea4ce4d48eb22e05f03b36e281149f1e944674fa)
Merged-In: I004843edf5a156dc07fb961edf80272f5cb50163
(cherry picked from commit d7f514abbe0bee01003ef694a67abfaa30667b4d)
---
 .../android/companiondevicemanager/DeviceChooserActivity.java  | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/packages/CompanionDeviceManager/src/com/android/companiondevicemanager/DeviceChooserActivity.java b/packages/CompanionDeviceManager/src/com/android/companiondevicemanager/DeviceChooserActivity.java
index e501e1269aeb..5ac059be2010 100644
--- a/packages/CompanionDeviceManager/src/com/android/companiondevicemanager/DeviceChooserActivity.java
+++ b/packages/CompanionDeviceManager/src/com/android/companiondevicemanager/DeviceChooserActivity.java
@@ -17,6 +17,7 @@
 package com.android.companiondevicemanager;

 import static android.companion.BluetoothDeviceFilterUtils.getDeviceMacAddress;
+import static android.view.WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS;

 import static java.util.Objects.requireNonNull;

@@ -58,6 +59,8 @@ public class DeviceChooserActivity extends Activity {
             Log.e(LOG_TAG, "About to show UI, but no devices to show");
         }

+        getWindow().addSystemFlags(SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
+
         if (getService().mRequest.isSingleDevice()) {
             setContentView(R.layout.device_confirmation);
             final DeviceFilterPair selectedDevice = getService().mDevicesFound.get(0);
--
2.29.0
